lib/api/core/validation/index.js

```diff
@@ -84,23 +84,23 @@ class Validation {
   }
 
   /**
+   * Validates a document against its collection specifications
+   *
    * @param {Request} request
    * @param {boolean} [verbose]
    * @returns {Promise.<{documentBody: *, errorMessages:string[], valid: boolean}|KuzzleRequest>}
    */
-  validationPromise(request, verbose = false) {
+  validate(request, verbose = false) {
     const
-      id = request.input.resource._id;
+      id = request.input.resource._id,
+      index = request.input.resource.index,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use destructuring syntax for these assignments from "request.input.resource". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3514)